libpod/container_internal.go

```diff
@@ -1866,8 +1866,11 @@ func (c *Container) cleanupStorage() error {
 		}
 	}
 	if c.config.RootfsMapping != nil {
-		if err := unix.Unmount(c.config.Rootfs, 0); err != nil {
-			logrus.Errorf("Unmounting idmapped rootfs for container %s after mount error: %v", c.ID(), err)
+		if err := unix.Unmount(c.config.Rootfs, 0); err != nil && err != unix.EINVAL {
+			if cleanupErr != nil {
+				logrus.Errorf("Unmounting idmapped rootfs for container %s after mount error: %v", c.ID(), err)
```

Comment:
Looking at the other code, this should be reporting the cleanupErr not the err?

Reply:
good point.  Thanks for noticing it.

I've copied the only instance in the function that was also broken.  Added a new patch to address that too